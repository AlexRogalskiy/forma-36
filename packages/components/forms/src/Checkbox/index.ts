export { Checkbox } from './Checkbox';
export type { CheckboxProps } from './Checkbox';
export type { CheckboxGroupProps } from './CheckboxGroup';
