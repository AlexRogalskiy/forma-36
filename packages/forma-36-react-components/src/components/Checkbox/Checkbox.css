@import 'settings/a11y';

.Checkbox {
  display: inline-block;
}

.Checkbox--disabled {
  cursor: not-allowed;
}
