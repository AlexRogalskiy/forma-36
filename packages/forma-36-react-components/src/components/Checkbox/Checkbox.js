import React from 'react';
import ControlledInput, {
  ControlledInputDefaultProps,
  ControlledInputPropTypes,
} from '../ControlledInput/ControlledInput';

const Checkbox = props => <ControlledInput type="checkbox" {...props} />;

Checkbox.propTypes = ControlledInputPropTypes;
Checkbox.defaultProps = {
  ...ControlledInputDefaultProps,
  testId: 'ctf-ui-checkbox',
};

export default Checkbox;
