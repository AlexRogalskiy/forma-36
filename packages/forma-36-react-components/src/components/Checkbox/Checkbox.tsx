import React, { FunctionComponent } from 'react';
import ControlledInput, {
  ControlledInputPropTypes,
} from '../ControlledInput/ControlledInput';

export const Checkbox: FunctionComponent<ControlledInputPropTypes> = props => (
  <ControlledInput type="checkbox" {...props} />
);

Checkbox.defaultProps = {
  isRequired: false,
  isDisabled: false,
  type: 'checkbox',
  testId: 'ctf-ui-checkbox',
};

export default Checkbox;
