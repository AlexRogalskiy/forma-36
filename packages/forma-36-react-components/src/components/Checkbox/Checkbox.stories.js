import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, boolean } from '@storybook/addon-knobs';
import { host } from 'storybook-host';
import { withInfo } from '@storybook/addon-info';

import Checkbox from './Checkbox';

storiesOf('Components|Checkbox', module)
  .addDecorator(
    host({
      align: 'center middle',
      cropMarks: false,
    }),
  )
  .add(
    'default',
    withInfo()(() => (
      <Checkbox
        extraClassNames={text('Extra Class Names', '')}
        id="Checkbox"
        checked={boolean('Checked', false)}
        labelText={(text('Aria label text'), 'some label text')}
        disabled={boolean('Disabled', false)}
        required={boolean('Required', false)}
        name={text('Name', 'some-name')}
      />
    )),
  );
