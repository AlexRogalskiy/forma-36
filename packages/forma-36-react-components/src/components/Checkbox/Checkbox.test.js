import React from 'react';
import { shallow, mount } from 'enzyme';
import { axe } from 'jest-axe';
import Checkbox from './Checkbox';

it('renders the component with all required props', () => {
  const output = shallow(<Checkbox id="checkbox" labelText="checkbox" />);

  expect(output).toMatchSnapshot();
});

it('renders the component with an additional class name', () => {
  const output = shallow(
    <Checkbox
      extraClassNames="my-extra-class"
      id="checkbox"
      labelText="checkbox"
    />,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component with required prop', () => {
  const output = shallow(
    <Checkbox
      extraClassNames="my-extra-class"
      id="checkbox"
      labelText="checkbox"
      required
    />,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component with disabled prop', () => {
  const output = shallow(
    <Checkbox
      extraClassNames="my-extra-class"
      id="checkbox"
      labelText="checkbox"
      disabled
    />,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component with as checked', () => {
  const output = shallow(
    <Checkbox
      extraClassNames="my-extra-class"
      id="checkbox"
      labelText="checkbox"
      checked
    />,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component with a value', () => {
  const output = shallow(
    <Checkbox
      extraClassNames="my-extra-class"
      id="checkbox"
      labelText="checkbox"
      value="someValue"
    />,
  );

  expect(output).toMatchSnapshot();
});

it('has no a11y issues', async () => {
  const output = mount(<Checkbox id="checkbox" labelText="checkbox" />).html();
  const results = await axe(output);

  expect(results).toHaveNoViolations();
});
