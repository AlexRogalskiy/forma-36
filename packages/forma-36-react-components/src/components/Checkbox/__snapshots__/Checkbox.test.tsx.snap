// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<ControlledInput
  isDisabled={false}
  labelText="checkbox"
  required={false}
  testId="ctf-ui-checkbox"
  type="checkbox"
/>
`;

exports[`renders the component with an additional class name 1`] = `
<ControlledInput
  className="my-extra-class"
  isDisabled={false}
  labelText="checkbox"
  required={false}
  testId="ctf-ui-checkbox"
  type="checkbox"
/>
`;
